service_capacity_modeling/capacity_planner.py

```diff
@@ -325,6 +325,92 @@ def models(self) -> Dict[str, CapacityModel]:
     def hardware_shapes(self) -> HardwareShapes:
         return self._shapes
 
+    def _plan_percentiles(
```

Comment:
Introduced this method to address the following issue.

Previously the steps to generate percentile plan was
* generate percentile desires
* plan_certain using the generated desires 
Issue in the above is, when percentile desires are generated the attributes that are not supplied by user are replaced with defaults declared in CapacityDesires interface. This results in `parent_desires.merge_with` not merging the defaults i.e. not merging the model defaults for attributes that are not supplied by user.

To mitigate the above issue, we first generate desires by merging with model defaults and then generate the percentile desires using the merged desires.

Reply:
Nice find